src/Forms/_FormValidationOverlay.test.js

```diff
@@ -0,0 +1,116 @@
+import { act } from 'react-dom/test-utils';
+import FormValidationOverlay from './_FormValidationOverlay';
+import { mount } from 'enzyme';
+import React from 'react';
+
+describe('<FormValidationOverlay />', () => {
+
+    const setup = (props = {}) => mount(
+        <FormValidationOverlay control={(<input onChange={() => {}} value='Test' />)} {...props} />);
+
+    const getPopover = () => {
+        return document.body.querySelector('.fd-popover__popper');
+    };
+
+    afterEach(() => {
+        document.body.innerHTML = '';
+    });
+
+    describe('default rendering', () => {
+        test('should render', () => {
+            let element = setup();
+
+            expect(element).toBeDefined();
+        });
+
+        test('should not render a FormMessage by default', () => {
+            let element = setup();
+
+            expect(element.find('FormMessage').length).toBe(0);
+        });
+    });
+
+    describe('Props', () => {
+        test('should allow props to be spread to the wrapping div component', () => {
+            let element = setup({ wrapperProps: { 'data-sample': 'Sample' } });
+
+            expect(
+                element.getDOMNode().attributes['data-sample'].value
+            ).toBe('Sample');
+        });
+
+        test('should set className on the outer div', () => {
+            const wrapper = setup({
+                className: 'wonderful-styles'
+            });
+
+            expect(
+                wrapper.find('.fd-popover').getDOMNode().classList
+            ).toContain('wonderful-styles');
+        });
+
+        test('should set className on the reference component', () => {
+            const wrapper = setup({
+                referenceClassName: 'wonderful-styles'
+            });
+
+            expect(
+                wrapper.find('.fd-popover__control').getDOMNode().classList
+            ).toContain('wonderful-styles');
+        });
+
+        test('should allow props to be spread to the FormMessage component', async() => {
+            let element;
+            await act(async() => {
+                element = setup({
+                    formMessageProps: { 'data-sample': 'Sample' },
+                    validationState: { state: 'error', message: 'test message' },
+                    show: true
+                });
+            });
+
+            expect(
+                element.find('FormMessage').getDOMNode().attributes['data-sample'].value
+            ).toBe('Sample');
+        });
+    });
+
+    describe('Interactions', () => {
```

Comment:
@skvale Looks like you solved this exact problem in tooltip in widgets recently - I copied and applied what you did there to get this working 🎉